src/App.js

```diff
@@ -6,42 +6,39 @@ import ChatLog from './components/ChatLog';
 import chatMessages from './data/messages.json';
 
 function App() {
-  const [entryData, setEntryData] = useState(chatMessages);
+  // const [entryData, setEntryData] = useState(chatMessages);
+  const [messages, setMessages] = useState(chatMessages);
+  const [likedMessages, setLikedMessages] = useState(new Set());
 
-  const [determineLikedState, setLikedState] = useState([]);
-  const addLiked = (indy) => {
-    setLikedState((oldState) => [...oldState, indy]);
-  };
-  const removeLiked = (indy) => {
-    setLikedState((oldState) => oldState.filter((item) => item !== indy));
-  };
-
-  const updateEntryData = (updatedEntry) => {
-    const entries = entryData.map((entry) => {
-      if (entry.id === updatedEntry.id) {
-        if (determineLikedState.size === 0) {
-          addLiked(entry.id);
-        } else if (determineLikedState.includes(entry.id)) {
-          removeLiked(entry.id);
-        } else {
-          addLiked(entry.id);
-        }
-        return updatedEntry;
+  const toggleLikeForMessage = (id) => {
```

Comment:
basically, the state in this App component is the source of truth, and all other components depend on this state to render their various dom elements and text.

the main functionality we want is for this state to change accordingly when the like button is pressed, which is what this function outlines. 

don't worry about this being a "heavy" (lots of lines of code) function. that's the way it should be in react - the logic (javascript) sits very closely with the html for every component. 

Now we think about what happens when the like button is pressed, purely in terms of state. The react components that we pass the state to will figure out what to render using the state. So in this case, when the like button is pressed, we want to update two things in the state:
- keep track of the like button press in the set of `likedMessages`
- toggle whether the message is in liked state or not liked state in `messages`

this function just does these two things